js-modules/defs.js

```diff
@@ -47,4 +47,9 @@ var JobT;
 var SourceSpansT;
 
 /** @define {boolean} */
+/* @ifndef RUN_PRETTIFY */
 var IN_GLOBAL_SCOPE = true;
+/* @endif */
+/* @ifdef RUN_PRETTIFY */
+var IN_GLOBAL_SCOPE = false;
```

Comment:
it won't because the file is never fed directly to closure compiler, it only sees `src/prettify.js` and `src/run_prettify.js` by which time those directives are already "processed" and only one of the two alternatives survives (either true or false depending on wheather `RUN_PRETTIFY` was set in the context (see the grunt preprocess task options)).

In fact the value used here doesn't really matter (I set it here only as a form of documentation), because we later pass the closure-compiler option `--define='IN_GLOBAL_SCOPE=??'` which overrides the const anyway: https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#define-type-description